### ISSUE TITLE
'Callable parameter usage violates definition' - false positive when type should be inferenced from 'static' type hint

### ISSUE THREAD

**Issue by 3DFace** [kalessil/phpinspectionsea#746]
Hi!

In example below, inspection (or IDE?) should inference a type of `Cat::create()` as a `Cat`. But it treats it as `static`. That leads to irrelevant warning.

```php
<?php

class Animal {

	/**
	 * @return static
	 */
	public static function create() {
		return new static();
	}

}

class Cat extends Animal {
}


class Example {

	/**
	 * @param Cat|null $cat
	 */
	public function acceptCat(?Cat $cat) : void {
		$cat = $cat ?? Cat::create(); // false positive: "New value type (static) is not in annotated types"
		echo $cat !== null;
	}

}
```

PhpStorm 2017.3.3 EAP
Build #PS-173.4301.15, built on January 10, 2018

**Comment by kalessil** [comment]
Thank you for reporting @3DFace, most probably it's a bug on our end.

**Comment by kalessil** [comment]
Fixed
